Sources/Network/Connection/Connections/ConsensusConnection.swift

```diff
@@ -32,35 +32,54 @@ final class ConsensusConnection:
 
         super.init(
             connectionOptionWrapperFactory: { transportProtocolOption in
+                let rotatedConfig = config.consensusConfig()
                 switch transportProtocolOption {
                 case .grpc:
                     return .grpc(
                         grpcService: grpcFactory.makeConsensusService(
-                            config: config,
+                            config: rotatedConfig,
                             targetQueue: targetQueue,
                             rng: rng,
                             rngContext: rngContext))
                 case .http:
                     return .http(httpService: httpFactory.makeConsensusService(
-                            config: config,
+                            config: rotatedConfig,
                             targetQueue: targetQueue,
                             rng: rng,
                             rngContext: rngContext))
                 }
             },
-            transportProtocolOption: config.transportProtocolOption,
+            transportProtocolOption: config.consensusConfig().transportProtocolOption,
             targetQueue: targetQueue)
     }
+    
+    func completionResultIntercept(
```

Comment:
lets factor this out into one place.

you could make this a function on Connection.swift that returns a closure with a signature same as this function, then at the call-site you dont need the closure syntax